zeebe/exporters/elasticsearch-exporter/src/main/java/io/camunda/zeebe/exporter/RestClientFactory.java

```diff
@@ -53,9 +65,56 @@ private HttpAsyncClientBuilder configureHttpClient(
       setupBasicAuthentication(config, builder);
     }
 
+    LOGGER.trace("Attempt to load interceptor plugins");
+    if (config.getInterceptorPlugins() != null) {
+      loadInterceptorPlugins(builder, config);
+    }
+
     return builder;
   }
 
+  private void loadInterceptorPlugins(
+      final HttpAsyncClientBuilder httpAsyncClientBuilder,
+      final ElasticsearchExporterConfiguration elsConfig) {
+    LOGGER.trace("Plugins detected to be not empty {}", elsConfig.getInterceptorPlugins());
+
+    final List<InterceptorPlugin> interceptors = elsConfig.getInterceptorPlugins();
+    for (final InterceptorPlugin interceptor : interceptors) {
+      LOGGER.trace("Attempting to register {}", interceptor.getId());
+      try {
+        // WARNING! Due to the nature of interceptors, by the moment they (interceptors)
+        // are executed, the below class loader will close the JAR file hence
+        // to avoid NoClassDefFoundError we must not close this class loader.
+        final var classLoader = ExternalJarClassLoader.ofPath(Paths.get(interceptor.getJarPath()));
+
+        final var pluginClass = classLoader.loadClass(interceptor.getClassName());
+        final var plugin = ReflectUtil.newInstance(pluginClass);
+
+        if (plugin instanceof final DatabaseCustomHeaderSupplier dchs) {
+          LOGGER.trace(
+              "Plugin {} appears to be a DB Header Provider. Registering with interceptor",
+              interceptor.getId());
+          httpAsyncClientBuilder.addInterceptorLast(
+              (HttpRequestInterceptor)
+                  (httpRequest, httpContext) -> {
+                    try {
+                      final var customHeader =
+                          ThreadContextUtil.callWithClassLoader(
+                              dchs::getElasticsearchCustomHeader, classLoader);
+                      httpRequest.addHeader(customHeader.key(), customHeader.value());
+                    } catch (final Exception e) {
+                      throw new RuntimeException(e);
+                    }
+                  });
+        } else {
+          throw new RuntimeException("Failed to load interceptor plugin");
```

Comment:
:x: Same comments as in the other modules/connectors (i.e. error handling, supplyWithClassLoader, etc.)